src/Modal.js

```diff
@@ -309,6 +315,8 @@ const Modal = React.createClass({
 
     this.setState(this._getStyles() //eslint-disable-line react/no-did-mount-set-state
       , () => this.focusModalContent());
+
+    this.needsFinalization = true;
```

Comment:
@jquense it was my initial though, but then I came into conclusion that explicit is better than implicit.